### PR TITLE
Fix duplication of jquery scripts

### LINKED ISSUE #372: Fix duplication of jquery script tags
Some layers seem to be creating a script tag with jquery('https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js') as its source and this is being appended to the head of the HTML file the source lives in. This seems to be creating a conflict here in plots2 just as the mapDependencies did. Here are the lines from one of the layers:
https://github.com/publiclab/leaflet-environmental-layers/blob/c69ad9a84b99734dc394b9ad34b6c23b3b67fb47/src/pfasLayer.js#L44-L46

Addressing this issue would solve the jquery related errors popping up in pages that contain maps.

The layers that create a new script tag for jquery are:
- pfasLayer.js
- toxicReleaseLayer.js
- osmLandfillMineQuarryLayer.js
- googleSpreadsheetLayer.js

_Originally posted by @crisner in https://github.com/publiclab/plots2/issues/7381#issuecomment-580131418_

Digging into LEL code...
The error happens on this line:
https://github.com/publiclab/leaflet-environmental-layers/blob/c69ad9a84b99734dc394b9ad34b6c23b3b67fb47/src/AllLayers.js#L220
And specifically only with toxicReleaseLayer (so far that I can tell) and only when it is on the screen. If that layer triggers whatever-it-is error, then no other maps will work right either, including sidebar maps.

🎈 Public Lab: Switzerland Fires - Google Chrome_031

🎈 Public Lab: Switzerland Fires - Google Chrome_032

🎈 Public Lab: Switzerland Fires - Google Chrome_034

I will take a look at why that particular layer is different from the others.

_Originally posted by @nstjean in https://github.com/publiclab/plots2/issues/7381#issuecomment-579945120_

### PR DESCRIPTION
**Pull request by crisner**
Fixes #372  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by nstjean** [comment]
So right now we're checking in every layer that needs it?

**Comment by crisner** [comment]
No. Just the ones that created the script tags from within. jQuery should be under the required dependencies in the readme for LEL.

**Comment by crisner** [comment]
So it is not really necessary to add a check. So I just fixed the files that were causing issues.

**Comment by nstjean** [comment]
Right. Which works! I just wonder if we can consolidate that code into just one spot? and run it once at the loading of LEL? Then we would know that jquery was installed for any layer that needs it.

**Comment by crisner** [comment]
Just confirmed. I have added the version of jQuery used by the layers under map dependencies in the readme.

**Comment by crisner** [comment]
> Right. Which works! I just wonder if we can consolidate that code into just one spot? and run it once at the loading of LEL? Then we would know that jquery was installed for any layer that needs it.

One place we could add that check would be in the AllLayers.js file. My concern is if we were to dynamically add the script we would need to wait for it to load to fetch layer data and the code for that is within those layer files.

**Comment by crisner** [comment]
One possible solution is we could add the check before even loading the layers into the overlays list.

**Comment by nstjean** [comment]
What about this... doesn't this already check for and install jquery?
https://github.com/publiclab/leaflet-environmental-layers/blob/86613f89be81b473921345e093eb69756c943603/src/leafletEnvironmentalLayers.js#L1

**Comment by nstjean** [comment]
But yes, I like the idea of checking before even loading the layers

**Comment by crisner** [comment]
The layers didn't work when I removed the script from the html file. These layers need version 1.7.1 of jQuery and I think the one added through require is a different version. Perhaps I could 'require' the version the layers need. This seems to be the cleaner way than adding the code to AllLayers.js file.

**Comment by nstjean** [comment]
If possible I think that would be ideal!

**Comment by jywarren** [comment]
Linking to similar discussion going on in another repo, by @sashadev-sky ! https://github.com/publiclab/Leaflet.DistortableImage/issues/538

**Comment by jywarren** [comment]
Thanks, all!

**Comment by crisner** [comment]
> Linking to similar discussion going on in another repo, by @sashadev-sky ! [publiclab/Leaflet.DistortableImage#538](https://github.com/publiclab/Leaflet.DistortableImage/issues/538)

Trying this out so that we could replace the jquery dependencies for all the layers.

**Comment by crisner** [comment]
Tried using XHR in one of the layers and it works without any issues. It got me thinking if I should try this using promises. Would it be ideal here? We are not chaining asynchronous calls where promises would come in handy. Should I use it anyway? What do you think @jywarren and @nstjean?

**Comment by nstjean** [comment]
I've actually never used promises before!

**Comment by crisner** [comment]
@sagarpreet-chadha, What do you think? Should I go ahead and change all layers to use xhr instead? Would it help in the long run?

**Comment by jywarren** [comment]
Ok, so this is a big one. Are we all good here, both @crisner and @nstjean ? Thank you!

**Comment by jywarren** [comment]
Ah, some comments hadn't loaded. OK, will wait on this! Thanks for this awesome work!

**Comment by nstjean** [comment]
I think if we can I'd like to merge this as long as it's working, and then we can continue to remove the jquery and simplify in another PR.

**Comment by crisner** [comment]
I have made this PR #383 which replaces the use of jquery in all files in the src directory. We still use jquery as a bootstrap dependency but we no longer use it to fetch data. I am not sure about a couple of files, waiting for feedback on it. 

This PR fixes only those files that caused a conflict in plots2. Depending on the priority we could merge this PR for now until the review for the other one is completed.

**Comment by nstjean** [comment]
Great! @sagarpreet-chadha Can we get this merged?

File: src/osmLandfillMineQuarryLayer.js [hunk 1]
```diff
@@ -35,51 +35,18 @@ L.LayerGroup.OSMLandfillMineQuarryLayer = L.LayerGroup.extend(
...
-        var script = document.createElement('SCRIPT');
-        script.src = 'https://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js';
-        script.type = 'text/javascript';
-        var northeast = self._map.getBounds().getNorthEast();
-        var southwest = self._map.getBounds().getSouthWest();
-
-        var currentMapZoom = self._map.getZoom();
-        if (currentMapZoom < info.OSMLandfillMineQuarryLayer.extents.minZoom) {
-          return;
+        if (typeof jQuery == 'undefined' || (typeof jQuery == 'function' && jQuery.fn.jquery !== '1.7.1')) {
```
Comment (sagarpreet-chadha) [inline src/osmLandfillMineQuarryLayer.js]: Why we are hardcoding 1.7.1 ? This is not required.
Any version of jquery works for us.

Comment (crisner) [reply src/osmLandfillMineQuarryLayer.js]: I am not sure but I tried removing jQuery 1.7.1 and jQuery 3 was available. It did not work.

Comment (jywarren) [reply src/osmLandfillMineQuarryLayer.js]: Can we try setting the version in the package.json file only, and managing compatibility that way? Thanks!

Comment (crisner) [reply src/osmLandfillMineQuarryLayer.js]: But wouldn't we need this version for bootstrap?
https://github.com/publiclab/leaflet-environmental-layers/blob/84da2d162eeeb0662be3c1e3de4c01a38086bf50/package.json#L53

Comment (crisner) [reply src/osmLandfillMineQuarryLayer.js]: It looks like I had added a slim version of jquery 3.3.1 which does not include the getJson function. Changed it to linking the full version and it works. :tada: I will be removing the lines that were responsible for duplicating the scripts and also the conditional statements I had added. Will also be removing references to the older version of jquery. Thanks @sagarpreet-chadha  and @jywarren!

**Comment by crisner** [comment]
@sagarpreet-chadha, @jywarren - Made the changes. Now we use the version of jquery available to fetch layer data. The test fixtures still refer to the older version of LEL. It shouldn't be a problem as it does not affect the layer data or the demo files in any way.

File: src/osmLandfillMineQuarryLayer.js (hunk 1 above)
Comment (sagarpreet-chadha) [reply src/osmLandfillMineQuarryLayer.js]: Awesome 🎉

**Pull request merged**